frontend/catalyst/jax_primitives.py

```diff
@@ -298,7 +298,10 @@ def _func_def_lowering(ctx, fn, call_jaxpr) -> str:
         # if supported and parameter-shift otherwise. Emulating the same behaviour
         # would require generating code to query the device.
         # For simplicity, Catalyst instead defaults to parameter-shift.
-        diff_method = fn.diff_method if fn.diff_method != "best" else "parameter-shift"
+        if fn.diff_method is None:
+            diff_method = "None"
+        else:
+            diff_method = "parameter-shift" if fn.diff_method == "best" else fn.diff_method
```

Comment:
Would this be sufficient as `str(None) == 'None'` and `str(str("")) == str("")` with almost no cost?
```suggestion
        diff_method = "parameter-shift" if fn.diff_method == "best" else str(fn.diff_method)
```